src/components/SingleTopic.jsx

```diff
@@ -0,0 +1,46 @@
+import { useParams } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { getArticles } from "../utils/api";
+
+const SingleTopic = () => {
+  const [articles, setArticles] = useState();
+  const [error, setError] = useState();
+  const { topic_slug } = useParams();
+
+  useEffect(() => {
+    getArticles()
+      .then((result) => {
+        setArticles(result);
+      })
+      .catch((err) => {
+        setError(err.response.data);
+      });
+  }, []);
+
+  if (!articles) {
+    return <h1>{error}</h1>;
+  }
+  return (
+    <>
+      <main>
+        <ul>
+          {articles.map((article) => {
+            if (article.topic === topic_slug)
```

Comment:
Instead of using getArticles and then filtering though the results, we can use the api itself to send back what we want. What happens if we pass topic to getArticles? Could you use your dependency array on line 18 to make it reusable?

Reply:
done